File: src/pick_and_place/scripts/read_rgb_image.py
#!/usr/bin/env python

import rospy
from sensor_msgs.msg import Image
from cv_bridge import CvBridge, CvBridgeError
import cv2

def image_callback(msg):
    # Initialize CvBridge
    bridge = CvBridge()
    
    try:
        # Convert the ROS Image message to OpenCV format
        cv_image = bridge.imgmsg_to_cv2(msg, "bgr8")
    except CvBridgeError as e:
        rospy.logerr("CvBridge Error: {0}".format(e))
        return

    # Display the image using OpenCV
    cv2.imshow("Camera Image", cv_image)
    
    # Add a delay to process OpenCV's window events
    cv2.waitKey(1)

def main():
    # Initialize the ROS node
    rospy.init_node('image_subscriber_node')

    # Subscribe to the image topic
    image_topic = "/camera/rgb/image_raw"  # Replace with the actual topic name
    rospy.Subscriber(image_topic, Image, image_callback)

    # Spin to keep the script running
    rospy.spin()

    # Clean up when done
    cv2.destroyAllWindows()

if __name__ == "__main__":
    main()

File: src/pick_and_place/src/read_rgb_image.cpp
#include <ros/ros.h>
#include <sensor_msgs/Image.h>
#include <cv_bridge/cv_bridge.h>
#include <opencv2/highgui/highgui.hpp>
#include <opencv2/imgproc/imgproc.hpp>

void imageCallback(const sensor_msgs::ImageConstPtr& msg) {
    try {
        // Convert the ROS Image message to an OpenCV image
        cv_bridge::CvImagePtr cv_image_ptr = cv_bridge::toCvCopy(msg, "bgr8");
        cv::Mat cv_image = cv_image_ptr->image;

        // Display the image using OpenCV
        cv::imshow("Camera Image", cv_image);
        cv::waitKey(1);
    } catch (cv_bridge::Exception& e) {
        ROS_ERROR("CvBridge Error: %s", e.what());
        return;
    }
}

int main(int argc, char** argv) {
    ros::init(argc, argv, "image_subscriber_node");
    ros::NodeHandle nh;

    // Subscribe to the image topic
    std::string image_topic = "/camera/rgb/image_raw";  // Replace with your actual topic name
    ros::Subscriber sub = nh.subscribe(image_topic, 10, imageCallback);

    // Spin to keep the node running
    ros::spin();

    // Clean up OpenCV windows
    cv::destroyAllWindows();

    return 0;
}
